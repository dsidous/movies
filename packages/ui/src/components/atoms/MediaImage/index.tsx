import React from 'react';

import useConfig from '../../hooks/useConfig';

interface Props {
  filePath: string;
  name: string;
  size: number;
  mediaType: string;
  className?: string;
}

const MediaImage: React.FC<Props> = ({
  filePath,
  name,
  size,
  mediaType,
  ...otherProps
}) => {
  const { getImageURL } = useConfig();

  const imageUrl = getImageURL({
    filePath,
    size,
    mediaType,
  });

  return <img src={imageUrl} alt={name} {...otherProps} />;
};

export default MediaImage;
