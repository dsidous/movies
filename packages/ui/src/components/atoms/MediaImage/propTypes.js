import { object, string, bool, number } from 'prop-types';

export const propTypes = {
  config: object,
  filePath: string,
  name: string,
  size: number,
  mediaType: string.isRequired,
  configLoading: bool,
};

export const defaultProps = {};
