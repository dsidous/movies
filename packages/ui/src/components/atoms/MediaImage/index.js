import React from 'react'
import { withConfig } from '@movies/common'

import { propTypes } from './propTypes'
import NoImage from '../../../images/noimage.jpg'

const MediaImage = ({
  config: { images },
  filePath,
  name,
  size,
  mediaType,
  configLoading,
  ...otherProps
}) => {
  if (configLoading) {
    return null
  }

  let imageUrl = NoImage

  if (filePath !== null) {
    const sizePath =
      mediaType === 'miniProfile'
        ? 'w132_and_h132_face/'
        : images[`${mediaType}_sizes`][size]

    imageUrl = `${images.secure_base_url}${sizePath}${filePath}`
  }

  return <img src={imageUrl} alt={name} {...otherProps} />
}

MediaImage.propTypes = propTypes

export default withConfig()(MediaImage)
