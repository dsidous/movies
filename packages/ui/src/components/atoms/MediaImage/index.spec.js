import React from 'react';
import { shallow } from 'enzyme';

import MediaImage from '.';

const mockProps = {
  config: {
    images: {
      secure_base_url: 'path/to/image',
      profile_sizes: ['1', '2', '3', '4'],
    },
  },
  filePath: 'path/to/image',
  name: 'name',
  size: 2,
  mediaType: 'profile',
  configLoading: true,
};

describe('Atoms/MediaImage', () => {
  it('should render as expected', () => {
    const wrapper = shallow(<MediaImage {...mockProps} />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should render with no filePath', () => {
    const wrapper = shallow(<MediaImage {...mockProps} filePath={null} />);

    expect(wrapper).toMatchSnapshot();
  });
});
