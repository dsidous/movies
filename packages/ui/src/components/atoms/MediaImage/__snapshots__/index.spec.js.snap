// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Atoms/MediaImage should render as expected 1`] = `ShallowWrapper {}`;

exports[`Atoms/MediaImage should render with no filePath 1`] = `ShallowWrapper {}`;
